lib/formats/dotnet-cli-parser.js

```diff
@@ -73,6 +72,8 @@ function parse(fileContent, packageTree) {
   // restructure tree with "from" chains.
   function buildPath(node, from) {
     node.from = node.from.concat(from);
+    node.from = _.uniq(node.from);
```

Comment:
Why is this needed? Why would the from path collected so far contain duplicates?

Reply:
strange to me too. when I add line 76 to include the current dep, it causes the previous dep to duplicate.
apparently the `forEach` is the culprit and I thought it cleaner to add `uniq` than `if` statements.
your thoughts?

Reply:
You are using concat, but Why are you not using `node.from.push(node.name + '@' + node.version)`?

Not that is explains the duplication, but let's not solve this with `uniq`. Let's get to the bottom of this. Add a test that reproduces this duplication over a fixture manifest file, use `devtool` to debug?

Reply:
I had to add the final `from` to the end of the recursive function